code/modules/lighting/lighting_object.dm

```diff
@@ -103,19 +103,18 @@
 	//anything that passes the first case is very likely to pass the second, and addition is a little faster in this case
 		icon_state = "transparent"
 		color = null
+	else if(!set_luminosity)
+		icon_state = "dark"
+		color = null
```

Comment:
I believe you don't need an icon state for dark tiles, just setting color to null and luminosity to 0 makes byond not render the tile